test/binaryen.js/kitchen-sink.js

```diff
@@ -887,6 +887,55 @@ function test_internals() {
   console.log('sizeof Literal: ' + Binaryen['_BinaryenSizeofLiteral']());
 }
 
+function test_for_each() {
+  module = new Binaryen.Module();
+
+  var v = module.addFunctionType("v", Binaryen.None, []);
+  var fn0 = module.addFunction("fn0", v, [], module.nop());
+  var fn1 = module.addFunction("fn1", v, [], module.nop());
+  var fn2 = module.addFunction("fn2", v, [], module.nop());
+
+  var i;
+  for (i = 0 ; i < module.getNumFunctions() ; i++) {
+    assert(module.getFunctionByIndex(i) === (0===i?fn0:(1===i?fn1:fn2)));
```

Comment:
Perhaps `fn0`, `fn1`, and `fn2` could be placed in an array `fns` and this could become `assert(module.getFunctionByIndex(i) === fns[i])`. I think that would be more readable. Similarly below.